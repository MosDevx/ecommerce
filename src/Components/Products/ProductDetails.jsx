import React from 'react';
import { useLocation } from 'react-router-dom';

const ProductDetails = () => {
  const location = useLocation();
  const product = location.state;

  return (
    <div className="lg:flex p-5">
      <div>
        <img src={product.image} alt={product.title} />
      </div>
      <div>
        <h2 className="font-extrabold text-xl my-4">{product.title}</h2>
        <p className="">{product.category}</p>
        <p className="font-extrabold my-3 text-xl">
          $
          {' '}
          {product.price}
          .00
        </p>
        <p>{product.description}</p>
        <button type="button" className="rounded-none bg-green-700 px-2 py-1 my-3 uppercase font-bold text-white">Add To Cart</button>
      </div>
    </div>
  );
};

export default ProductDetails;
