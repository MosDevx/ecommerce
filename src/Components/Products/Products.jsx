import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { searchProduct } from '../../Redux/ProductSlice';

const Products = ({ products }) => {
  const dispatch = useDispatch();
  return (
    <div>
      <form className="w-full">
        <label
          htmlFor="default-search"
          className="mb-2 text-sm font-medium text-gray-900 sr-only dark:text-gray-300"
        >
          Search
        </label>
        <div className="relative">
          <input
            type="search"
            id="default-search"
            className="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-full border border-gray-300 focus:ring-gray-500 focus:border-gray-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-gray-500 dark:focus:border-gray-500"
            placeholder="Search..."
            required
            onChange={(e) => { dispatch(searchProduct(e.target.value)); }}
          />
          <div className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none">
            <svg className="w-5 h-5 text-gray-500 dark:text-gray-400" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z" /></svg>
          </div>
        </div>
      </form>

      <h3 className="m-4">
        Total Stocks:
        {' '}
        {products.length}
      </h3>
      <div className="grid grid-cols-2">
        {products && products.map((product) => (
          <div key={product.id} className="basis-1/4 flex justify-center flex-col text-center items-center p-2 cursor-pointer hover:border border-gray-700">
            <NavLink to={`/product/${product.id}`} state={product}>
              <img
                className="object-contain w-[200px] aspect-square object-contain"
                src={product.image}
                alt="picture1"
              />
              <div className="mt-4 flex flex-col items-center">
                <p className="font-bold">{product.category}</p>
                <p>{product.title}</p>
                <p className="font-extrabold">
                  $
                  {' '}
                  {product.price}
                </p>
              </div>
            </NavLink>
          </div>
        )) }
      </div>
    </div>
  );
};

export default Products;
