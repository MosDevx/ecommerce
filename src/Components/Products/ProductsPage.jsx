import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getProductsData } from '../../Redux/ProductSlice';
import Products from './Products';

const ProductsPage = () => {
  const products = useSelector((state) => state.products);
  const dispatch = useDispatch();

  console.log('Products page', products.products);

  useEffect(() => {
    if (products.products.length === 0) {
      dispatch(getProductsData());
    }
  }, [dispatch, products.products.length], products.products.length);
  return (
    products.status === 'Loading' ? <h2>Loading</h2> : <Products products={products.filteredProduct} key={products.filteredProduct.id} />
  );
};

export default ProductsPage;
