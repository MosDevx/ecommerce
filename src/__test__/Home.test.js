import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Products from '../Components/Products/Products';
import ProductsPage from '../Components/Products/ProductsPage';

jest.mock('react-redux');
jest.mock('react-router-dom');

describe('Test Home component rendering', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Test product component rendering on homepage', () => {
    useSelector.mockReturnValue({
      products: [
        {
          category: 'electronics',
          description: 'test',
          id: 14,
          image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
          price: 999.99,
          title: 'title',
          rating: {
            rate: 2.2,
            count: 140,
          },
        },
      ],
    });
    const tree = render(<Products />);
    expect(tree).toMatchSnapshot();
  });

  test('Test product page component rendering', () => {
    useSelector.mockReturnValue({
      products: [
        {
          category: 'electronics',
          description: 'test',
          id: 14,
          image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
          price: 999.99,
          title: 'title',
          rating: {
            rate: 2.2,
            count: 140,
          },
        },
      ],
    });
    const tree = render(<ProductsPage />);
    expect(tree).toMatchSnapshot();
  });
});
