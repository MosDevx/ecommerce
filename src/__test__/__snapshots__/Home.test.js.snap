// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home component rendering Test product component rendering on homepage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mx-5"
      >
        <form
          class="w-full"
        >
          <div
            class="relative"
          >
            <input
              class="block p-3 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-full border border-gray-300 focus:ring-gray-500 focus:border-gray-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-gray-500 dark:focus:border-gray-500"
              id="default-search"
              placeholder="Search..."
              required=""
              type="search"
            />
            <div
              class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
            >
              <svg
                class="w-5 h-5 text-gray-500 dark:text-gray-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
          </div>
        </form>
        <div
          class="grid grid-cols-2"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mx-5"
    >
      <form
        class="w-full"
      >
        <div
          class="relative"
        >
          <input
            class="block p-3 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-full border border-gray-300 focus:ring-gray-500 focus:border-gray-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-gray-500 dark:focus:border-gray-500"
            id="default-search"
            placeholder="Search..."
            required=""
            type="search"
          />
          <div
            class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
          >
            <svg
              class="w-5 h-5 text-gray-500 dark:text-gray-400"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
        </div>
      </form>
      <div
        class="grid grid-cols-2"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
